The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/7177767745/job/19544782892) [(plain)](https://github.com/rust-lang/rust/commit/4d38e8aee6b358c52651c0efcd0e395830ebbef7/checks/19544782892/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
   Compiling libc v0.2.150
   Compiling memchr v2.5.0
   Compiling std v0.0.0 (/checkout/library/std)
   Compiling compiler_builtins v0.1.103
error[E0599]: no method named `to_u32` found for reference `&i8` in the current scope
   --> library/core/src/fmt/num.rs:478:71
    |
478 |     impl_Display!(i8, u8, i16, u16, i32, u32, isize, usize as u32 via to_u32 named fmt_u32);
    |                                                                       ^^^^^^ method not found in `&i8`

error[E0599]: no method named `to_u32` found for reference `&u8` in the current scope
   --> library/core/src/fmt/num.rs:478:71
    |
478 |     impl_Display!(i8, u8, i16, u16, i32, u32, isize, usize as u32 via to_u32 named fmt_u32);
    |                                                                       ^^^^^^ method not found in `&u8`

error[E0599]: no method named `to_u32` found for reference `&i16` in the current scope
   --> library/core/src/fmt/num.rs:478:71
    |
478 |     impl_Display!(i8, u8, i16, u16, i32, u32, isize, usize as u32 via to_u32 named fmt_u32);
    |                                                                       ^^^^^^ method not found in `&i16`

error[E0599]: no method named `to_u32` found for reference `&u16` in the current scope
   --> library/core/src/fmt/num.rs:478:71
    |
478 |     impl_Display!(i8, u8, i16, u16, i32, u32, isize, usize as u32 via to_u32 named fmt_u32);
    |                                                                       ^^^^^^ method not found in `&u16`

error[E0599]: no method named `to_u32` found for reference `&i32` in the current scope
   --> library/core/src/fmt/num.rs:478:71
    |
478 |     impl_Display!(i8, u8, i16, u16, i32, u32, isize, usize as u32 via to_u32 named fmt_u32);
    |                                                                       ^^^^^^ method not found in `&i32`

error[E0599]: no method named `to_u32` found for reference `&u32` in the current scope
   --> library/core/src/fmt/num.rs:478:71
    |
478 |     impl_Display!(i8, u8, i16, u16, i32, u32, isize, usize as u32 via to_u32 named fmt_u32);
    |                                                                       ^^^^^^ method not found in `&u32`

error[E0599]: no method named `to_u32` found for reference `&isize` in the current scope
   --> library/core/src/fmt/num.rs:478:71
    |
478 |     impl_Display!(i8, u8, i16, u16, i32, u32, isize, usize as u32 via to_u32 named fmt_u32);
    |                                                                       ^^^^^^ method not found in `&isize`

error[E0599]: no method named `to_u32` found for reference `&usize` in the current scope
   --> library/core/src/fmt/num.rs:478:71
    |
478 |     impl_Display!(i8, u8, i16, u16, i32, u32, isize, usize as u32 via to_u32 named fmt_u32);
    |                                                                       ^^^^^^ method not found in `&usize`

error[E0599]: no method named `to_u32` found for reference `&i8` in the current scope
   --> library/core/src/fmt/num.rs:480:67
    |
480 |     impl_Exp!(i8, u8, i16, u16, i32, u32, isize, usize as u32 via to_u32 named exp_u32);
    |                                                                   ^^^^^^ method not found in `&i8`

error[E0599]: no method named `to_u32` found for reference `&u8` in the current scope
   --> library/core/src/fmt/num.rs:480:67
    |
480 |     impl_Exp!(i8, u8, i16, u16, i32, u32, isize, usize as u32 via to_u32 named exp_u32);
    |                                                                   ^^^^^^ method not found in `&u8`

error[E0599]: no method named `to_u32` found for reference `&i16` in the current scope
   --> library/core/src/fmt/num.rs:480:67
    |
480 |     impl_Exp!(i8, u8, i16, u16, i32, u32, isize, usize as u32 via to_u32 named exp_u32);
    |                                                                   ^^^^^^ method not found in `&i16`

error[E0599]: no method named `to_u32` found for reference `&u16` in the current scope
   --> library/core/src/fmt/num.rs:480:67
    |
480 |     impl_Exp!(i8, u8, i16, u16, i32, u32, isize, usize as u32 via to_u32 named exp_u32);
    |                                                                   ^^^^^^ method not found in `&u16`

error[E0599]: no method named `to_u32` found for reference `&i32` in the current scope
   --> library/core/src/fmt/num.rs:480:67
    |
480 |     impl_Exp!(i8, u8, i16, u16, i32, u32, isize, usize as u32 via to_u32 named exp_u32);
    |                                                                   ^^^^^^ method not found in `&i32`

error[E0599]: no method named `to_u32` found for reference `&u32` in the current scope
   --> library/core/src/fmt/num.rs:480:67
    |
480 |     impl_Exp!(i8, u8, i16, u16, i32, u32, isize, usize as u32 via to_u32 named exp_u32);
    |                                                                   ^^^^^^ method not found in `&u32`

error[E0599]: no method named `to_u32` found for reference `&isize` in the current scope
   --> library/core/src/fmt/num.rs:480:67
    |
480 |     impl_Exp!(i8, u8, i16, u16, i32, u32, isize, usize as u32 via to_u32 named exp_u32);
    |                                                                   ^^^^^^ method not found in `&isize`

error[E0599]: no method named `to_u32` found for reference `&usize` in the current scope
   --> library/core/src/fmt/num.rs:480:67
    |
480 |     impl_Exp!(i8, u8, i16, u16, i32, u32, isize, usize as u32 via to_u32 named exp_u32);
    |                                                                   ^^^^^^ method not found in `&usize`
For more information about this error, try `rustc --explain E0599`.
error: could not compile `core` (lib) due to 16 previous errors
warning: build failed, waiting for other jobs to finish...
Build completed unsuccessfully in 0:00:07
```

</details>

--- a/library/core/src/fmt/num.rs
+++ b/library/core/src/fmt/num.rs
@@ -15,7 +15,7 @@ trait DisplayInt:
     fn zero() -> Self;
     fn from_u8(u: u8) -> Self;
     fn to_u8(&self) -> u8;
-    fn to_u16(&self) -> u16;
+    #[cfg(not(any(target_pointer_width = "64", target_arch = "wasm32")))]
     fn to_u32(&self) -> u32;
     fn to_u64(&self) -> u64;
     fn to_u128(&self) -> u128;
@@ -27,7 +27,7 @@ macro_rules! impl_int {
           fn zero() -> Self { 0 }
           fn from_u8(u: u8) -> Self { u as Self }
           fn to_u8(&self) -> u8 { *self as u8 }
-          fn to_u16(&self) -> u16 { *self as u16 }
+          #[cfg(not(any(target_pointer_width = "64", target_arch = "wasm32")))]
           fn to_u32(&self) -> u32 { *self as u32 }
           fn to_u64(&self) -> u64 { *self as u64 }
           fn to_u128(&self) -> u128 { *self as u128 }
@@ -40,7 +40,7 @@ macro_rules! impl_uint {
           fn zero() -> Self { 0 }
           fn from_u8(u: u8) -> Self { u as Self }
           fn to_u8(&self) -> u8 { *self as u8 }
-          fn to_u16(&self) -> u16 { *self as u16 }
+          #[cfg(not(any(target_pointer_width = "64", target_arch = "wasm32")))]
           fn to_u32(&self) -> u32 { *self as u32 }
           fn to_u64(&self) -> u64 { *self as u64 }
           fn to_u128(&self) -> u128 { *self as u128 }

